core/domain/exp_services.py

```diff
@@ -1864,24 +1865,30 @@ def is_version_of_draft_valid(exp_id, version):
 def get_user_exploration_data(
         user_id, exploration_id, apply_draft=False, version=None):
     """Returns a description of the given exploration."""
+    exp_user_data = user_models.ExplorationUserDataModel.get(
+        user_id, exploration_id)
     if apply_draft:
-        exploration = get_exp_with_draft_applied(exploration_id, user_id)
+        exploration = (
```

Comment:
Let's maybe call this updated_exploration, and then set exploration to updated_exploration in the else case, so that each variable represents a clear concept. (I.e. updated_exploration is the thing you get from applying the draft, and exploration is the thing you use subsequently.)

Reply:
Sounds good.

Reply:
Done.